### PR TITLE
Change python package installation policy

### PR DESCRIPTION
**Pull request by jngrad**
Install from apt-get instead of pip whenever possible to avoid replacing python packages the OS is relying on. Don't compile and install untrusted/unstable packages with root rights for security reasons.

### CODE REVIEW

File: docker/ubuntu-python3/Dockerfile-cuda-10.1 [hunk 1]
```diff
@@ -58,4 +55,7 @@ RUN tlmgr init-usertree \
 && tlmgr install lm stmaryrd 2>&1 || : \
 && updmap -user
 
+# install contributed Python3 packages
+RUN pip3 install --user --upgrade sphinxcontrib-bibtex 'MDAnalysis>=0.16' 'pint>=0.9'
```
Comment (KaiSzuttor) [inline docker/ubuntu-python3/Dockerfile-cuda-10.1]: have you checked for a package for pint?

Comment (jngrad) [reply docker/ubuntu-python3/Dockerfile-cuda-10.1]: I had only checked pint's website for the installation instructions, which don't advertise `python3-pint`. Silly me.

Comment (jngrad) [reply docker/ubuntu-python3/Dockerfile-cuda-10.1]: `python3-pint` is currently version 0.8.1-2 (from 2017) in Ubuntu 18.04, we should probably stick with 0.9 from pip which has fixes for numpy-related compatibility bugs.

**Comment by jngrad** [comment]
A more general question on security: should we enforce using `pip install` only as user? Installing a package with pip as root can pull in many dependencies, potentially updating or overshadowing one installed from apt-get. For example `MDAnalysis`, currently installed as user, installs locally a different version of numpy, scipy, matplotlib, setuptools and sphinx.

**Comment by jngrad** [comment]
@RudolfWeeber here's the result of installing `MDAnalysis` as user:
```
Successfully built MDAnalysis
Installing collected packages: six, numpy, scipy, GridDataFormats, biopython, gsd,
   joblib, cycler, python-dateutil, setuptools, kiwisolver, pyparsing, matplotlib,
   msgpack, mmtf-python, mock, decorator, networkx, MDAnalysis
Successfully installed GridDataFormats-0.5.0 MDAnalysis-0.20.1 biopython-1.74
   cycler-0.10.0 decorator-4.4.0 gsd-1.9.3 joblib-0.14.0 kiwisolver-1.1.0
   matplotlib-3.1.1 mmtf-python-1.1.2 mock-3.0.5 msgpack-0.6.2 networkx-2.4 numpy-1.17.3
   pyparsing-2.4.2 python-dateutil-2.8.0 scipy-1.3.1 setuptools-41.4.0 six-1.12.0
```
When installing as root, the only difference is that `cycler` is not installed (for all other packages, the version number are the same as in the user install). In my previous message, the `sphinx` package actually came from `sphinxcontrib-bibtex`.

To install packages with version pinning, we must have only one pip install command, right?

**Comment by jngrad** [comment]
espressomd/espresso#3194 depends on this PR

**Pull request merged**